examples/basic/variables.tf

```diff
@@ -2,32 +2,35 @@
 # Input variables
 ########################################################################################################################
 
-variable "ibmcloud_api_key" {
+variable "resource_group" {
   type        = string
-  description = "The IBM Cloud API Key"
-  sensitive   = true
+  description = "An existing resource group name to use for this example. If not specified, a new resource group is created."
+  default     = null
 }
 
-variable "region" {
+variable "ibmcloud_api_key" {
+  description = "Used with the Terraform IBM-Cloud/ibm provider"
+  sensitive   = true
   type        = string
-  description = "Region to provision all resources created by this example"
-  default     = "us-south"
 }
 
-variable "prefix" {
+variable "region" {
+  default     = "us-south"
+  description = "Used with the Terraform IBM-Cloud/ibm provider as well as resource creation."
   type        = string
-  description = "Prefix to append to all resources created by this example"
-  default     = "basic"
+  validation {
```

Comment:
I wouldn't add validation to the example - just add it to the module, otherwise you need to maintain in 2 places